web/src/search_suggestion.ts

```diff
@@ -60,33 +85,31 @@ function check_validity(last, terms, valid, invalid) {
     return true;
 }
 
-function format_as_suggestion(terms) {
+function format_as_suggestion(terms: NarrowTerm[]): Suggestion {
     return {
         description_html: Filter.search_description_as_html(terms),
         search_string: Filter.unparse(terms),
     };
 }
 
-function compare_by_huddle(huddle) {
-    huddle = huddle.slice(0, -1).map((person) => {
-        person = people.get_by_email(person);
-        return person && person.user_id;
+function compare_by_huddle(huddle: string[]): (person1: User, person2: User) => number {
+    const user_ids = huddle.slice(0, -1).flatMap((person) => {
+        const user = people.get_by_email(person);
+        return user?.user_id ?? [];
```

Comment:
Can you do the rename commit first so that you don't have to rename the variables here too?